submissions/pozzitive11/memory-pair-game/js/index.js

```diff
@@ -0,0 +1,125 @@
+const cards = [
+  {
+    front: "images/front.jpg",
+    back: "images/sleepy-cat.jpg",
+    name: "sleepy",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/cat.jpg",
+    name: "cat",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/frightened-cat.jpg",
+    name: "frightened",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/brutal-cat.jpg",
+    name: "brutal",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/rock-cat.jpg",
+    name: "rock",
+  },
+  {
+    front: "images/front.jpg",
+    back: "images/cute-cat.jpg",
+    name: "cute",
+  },
+];
+
+const gameCards = cards.map(
+  ({ front, back, name }) =>
+    `<div class="game__card" id="${name}">
+             <img class="front__img" src="${front}" alt="">
+             <img class="back__img" src="${back}" alt="">
+           </div>
+           <div class="game__card" id="${name}">
+             <img class="front__img" src="${front}" alt="">
+             <img class="back__img" src="${back}" alt="">
+           </div>`
+);
+
+const gameContainer = document.querySelector(".game__container");
+gameContainer.innerHTML = gameCards.join("");
+const allGameCards = document.querySelectorAll(".game__card");
+
+let numberOfMatches = 0;
+let flippedCards = [];
+
+gameContainer.addEventListener("click", ({ target }) => {
+  const parentTarget = target.closest(".game__card");
+  if (parentTarget !== null) {
+    parentTarget.classList.add("flip");
+
+    flippedCards.push(parentTarget);
+    checkArrayOfFlippedCardsLength();
+    checkForWin();
+  }
+});
+
+function checkArrayOfFlippedCardsLength() {
+  if (flippedCards.length === 2) {
+    checkMatch();
+  }
+}
+
+function checkMatch() {
+  const match = flippedCards[0].id === flippedCards[1].id;
+
+  if (match) {
+    numberOfMatches++;
+    addingDisableClass();
+    flippedCards = [];
+    setTimeout(() => {
+      addingDisableClass();
+    }, 1000);
+  } else {
+    removeFlip();
+  }
+}
+
+function removeFlip() {
+  setTimeout(() => {
+    flippedCards[0].classList.remove("flip");
+    flippedCards[1].classList.remove("flip");
+    flippedCards = [];
+
+    addingDisableClass();
+
+    checkForWin();
+  }, 1000);
+  addingDisableClass();
+}
+
+function addingDisableClass() {
+  allGameCards.forEach((item) => {
```

Comment:
Instead add it into field itself. In the place where your event listener are placed

Reply:
I don't really understand the meaning of this action, can you please explain

Reply:
Sure. You can add `disabled-game-container` instead of `disabled-click`. And then check this class in your event listener.

Reply:
but I use this function in other functions

Reply:
done